### PR TITLE
Add method ActionCable::Channel#stop_stream_from, ActionCable::Channe…

### PR DESCRIPTION
**Pull request by piecehealth**
Add methods `ActionCable::Channel#stop_stream_from`, `ActionCable::Channel#stop_stream_for` to allow user unsubscribe specify steam.

### Summary

Users can follow multiple streams in one channel by calling `stream_from` or `stream_for` multiple times. As far as I know, `ActionCable` only provide `ActionCable::Channel#stop_all_streams` to unfollow all streams, So I add two methods to allow user unsubscribe specify steam.

### CODE REVIEW

File: actioncable/test/channel/stream_test.rb [hunk 1]
```diff
@@ -144,6 +144,116 @@ class StreamTest < ActionCable::TestCase
...
+      end
+    end
+
+    private
+      def subscribers_of(connection)
+        connection
+          .pubsub
+          .subscriber_map
+          .instance_variable_get(:@subscribers)
+      end
```
Comment (jeremy) [inline actioncable/test/channel/stream_test.rb]: Testing against internal implementation details is brittle, but feels appropriate in this case. Alternatively, we'd need to broadcast and check which subscribers received it.

File: actioncable/lib/action_cable/channel/streams.rb [hunk 2]
```diff
@@ -82,7 +82,7 @@ def stream_from(broadcasting, callback = nil, coder: nil, &block)
         # Build a stream handler by wrapping the user-provided callback with
         # a decoder or defaulting to a JSON-decoding retransmitter.
         handler = worker_pool_stream_handler(broadcasting, callback || block, coder: coder)
-        streams << [ broadcasting, handler ]
+        streams[broadcasting] = [ broadcasting, handler ]
```
[the remark this answers is not in the document]
Comment (piecehealth) [reply actioncable/lib/action_cable/channel/streams.rb]: Good point!
I didn't see any test cases to cover this scenario: user can subscribe to the same pubsub broadcasting with multiple subscriptions, and I think it make sense to overwrite the previous subscription with a new one, just like we can define methods with the same name multiple times, but only last one will work.

What's your opinion? 😉

File: actioncable/test/channel/stream_test.rb (hunk 1 above)
Comment (piecehealth) [reply actioncable/test/channel/stream_test.rb]: Agree, maybe it's possible to switch test pubsub adapter from `actioncable/test/stubs/test_adapter.rb` to `actioncable/lib/action_cable/subscription_adapter/inline.rb`

**Comment by piecehealth** [comment]
Hi @jeremy ,
I did some changes based on your comments, would you mind to take another look 😉 ?

**Review by jeremy — APPROVED** [review]

**Pull request merged**